microservices/restful-api-design-guidelines.md

```diff
@@ -0,0 +1,45 @@
+# RESTful API Design Guidelines
+
+## Introduction
+This page documents some guidelines to be used when designing and specifying RESTful APIs for microservices within the Survey Data Collection (SDC) platform.
+
+## HATEOAS
+We will not be building full [HATEOAS](https://en.wikipedia.org/wiki/HATEOAS)-complaint APIs as there is currently no clear requirement for the SDC platform that justifies the additional development complexity and effort.
+
+## URIs
+URIs should be all lowercase. The primary means for identifying a resource to access is via its UUID, as per the object identity strategy. The UUID should appear after the plural resource name without further qualification:
+
+`GET /collectionexercises/2c82b85a-2a03-434d-8019-eae05b01c6f4`
+
+Where a secondary means of access is supported, it must be further qualified with the name of the non-ID attribute:
+
+`GET /parties/ref/49900000001`
+
+Resource names in URIs should be pluralised, apart from where the identity of a secondary resource is used to further qualify the request, in which case the secondary resource name should be singular. For example, retrieving the collection exercises for a survey:
+
+`GET /collectionexercises/survey/cb0711c3-0ac8-41d3-ae0e-567e5a1ef87`
+
+## HTTP Status Codes
+Endpoints should return the appropriate HTTP status code. Including, but not limited to:
+
+* `HTTP 201 Created` - when a resource has been created
```

Comment:
Not sure you need to list the codes - should just be able to state that correct codes should be returned as appropriate and potentially link to a resource list of standard HTTP codes

Reply:
OK.